static/js/apps/eval_rig/feedback_form.tsx

```diff
@@ -37,144 +53,202 @@ export interface EvalInfo {
+  }, [sheetId, sessionQueryId, sessionCallId]);
+
+  useEffect(() => {
+    const sheet = doc.sheetsByTitle[DC_CALL_SHEET];
+    if (!(sessionQueryId in allCall)) {
+      setEvalInfo(null);
+      return;
+    }
+    const rowIdx = allCall[sessionQueryId][sessionCallId];
+    sheet.getRows({ offset: rowIdx - 1, limit: 1 }).then((rows) => {
+      const row = rows[0];
+      if (row) {
+        setEvalInfo({
+          dcResponse: row.get(DC_RESPONSE_COL),
+          dcStat: row.get(DC_STAT_COL),
+          llmStat: row.get(LLM_STAT_COL),
+          question: row.get(DC_QUESTION_COL),
+        });
+      }
+    });
+  }, [doc, allCall, sessionQueryId, sessionCallId, setSessionCallId]);
 
   const handleChange = (event) => {
     const { name, value } = event.target;
-    setResponse((prevState) => ({
-      ...prevState,
-      [name]: value,
-    }));
+    setResponse((prevState) => {
+      const newState = {
+        ...prevState,
+        [name]: value,
+      };
+      let tmpStatus = FormStatus.Completed;
+      for (const value of Object.values(newState)) {
+        if (value === "") {
+          tmpStatus = FormStatus.InProgress;
+          break;
+        }
+      }
+      setStatus(tmpStatus);
+      return newState;
+    });
   };
 
-  const handleSubmit = (event) => {
+  const handleSubmit = async (event): Promise<boolean> => {
     event.preventDefault();
-    response.userEmail = userEmail;
-    saveToStore(sheetId, props.queryId, props.callId, response);
-    saveToSheet(doc, props.queryId, props.callId, response);
+    loadSpinner(LOADING_CONTAINER_ID);
+    return Promise.all([
+      saveToStore(userEmail, sheetId, sessionQueryId, sessionCallId, response),
+      saveToSheet(userEmail, doc, sessionQueryId, sessionCallId, response),
+    ])
+      .then(() => {
+        return true;
+      })
+      .catch((error) => {
+        alert("Error submitting response: " + error);
+        return false;
+      })
+      .finally(() => {
+        removeSpinner(LOADING_CONTAINER_ID);
+      });
   };
 
+  useImperativeHandle(ref, () => ({
+    status: () => {
+      return status;
+    },
+    submitForm: async (): Promise<boolean> => {
+      return handleSubmit(new Event("submit"));
+    },
+  }));
+
   let dcResponseOptions;
-  if (props.evalInfo.dcStat) {
-    dcResponseOptions = {
-      DC_ANSWER_IRRELEVANT: "Doesn't match the question",
-      DC_ANSWER_RELEVANT: "Relevant and direct",
-    };
-  } else {
-    dcResponseOptions = {
-      DC_ANSWER_EMPTY_BADNL: "Data exists, but NL fails to respond",
-      DC_ANSWER_EMPTY_NODATA: "Query asks for data that doesn't exist in DC",
-      DC_ANSWER_EMPTY_OUTOFSCOPE:
-        "Query asks for data that is out-of-scope for DC",
-    };
+  if (evalInfo) {
+    if (evalInfo.dcStat) {
+      dcResponseOptions = {
+        DC_ANSWER_IRRELEVANT: "Doesn't match the question",
+        DC_ANSWER_RELEVANT: "Relevant and direct",
+      };
+    } else {
+      dcResponseOptions = {
```

Comment:
can we make these options objects into constants so there is one place to see what all the options are